lib/scholar/cluster/gmm.ex

```diff
@@ -0,0 +1,433 @@
+defmodule Scholar.Cluster.GaussianMixture do
+  @moduledoc """
+  Gaussian Mixture Model.
+
+  Gaussian Mixture Model is a probabilistic model that assumes every data point is generated
+  by choosing one of several fixed Gaussian distributions and then sampling from it.
+  Its parameters are estimated using the Expectation-Maximization (EM) algorithm, which is an
+  iterative algorithm alternating between the two steps: the E-step which computes the
+  expectation of the Gaussian assignment for each data point x and the M-step which updates the
+  parameters to maximize the expectations found in E-step. While every iteration of the algorithm
+  is guaranteed to improve the log-likelihood, the final result depends on the initial values of
+  the parameters and the entire procedure should be repeated several times.
```

Comment:
```suggestion
  the parameters. Thus, the algorithm consists of repeating this procedure several times to find the best result. 
```

WDYT? @msluszniak @krstopro

Reply:
I would go for "Thus the procedure consists of repeating the algorithm several times and taking the best obtained result" as "algorithm" refers to EM algorithm mentioned before.

Reply:
```suggestion
  the parameters. Thus the procedure consists of repeating the algorithm several times
  and taking the best obtained result.
```